### PR TITLE
Expose CA flags `new-pod-scaleup-delay` and `max-empty-bulk-delete` 

### LINKED ISSUE #252: Expose flags to control scale-up and scale-down in autoscaler
**What would you like to be added**:

The following flags which are provided by g/autoscaler should be exposed on the `shoot Yaml`
1) [new-pod-scaleup-delay](https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/FAQ.md#how-can-i-modify-cluster-autoscaler-reaction-time)
2) [max-empty-bulk-delete](https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/FAQ.md#how-can-i-modify-cluster-autoscaler-reaction-time:~:text=Empty%20nodes%2C%20on%20the%20other%20hand%2C%20can%20be%20terminated%20in%20bulk%2C%20up%20to%2010%20nodes%20at%20a%20time%20(configurable%20by%20%2D%2Dmax%2Dempty%2Dbulk%2Ddelete%20flag. )

**Why is this needed**:
To allow customer more control on scale-up , scale-down of CA
Refer live issue # 3735

### PR DESCRIPTION
**Pull request by aaronfern**
**How to categorize this PR?**

/area auto-scaling
/kind enhancement api-change

**What this PR does / why we need it**:
This PR updated the shoot API to adds two new fields to `.spec.kubernetes.clusterAutoscaler`
1. NewPodScaleupDelay
2. MaxEmptyBulkDelete

These fields are to be passed onto `cluster-autoscaler` as configurable flags and are meant to help users have more control on the scale-up and scale-down of their machine deployments

**Which issue(s) this PR fixes**:
Fixes gardener/autoscaler#252

**Special notes for your reviewer**:

**Release note**:

```feature user
Machine scale-up delay for new pods can now be configured for `cluster-autoscaler` via the field `.spec.kubernetes.clusterAutoscaler.newPodScaleupDelay` in the `Shoot` API .
```
```feature user
Concurrent empty machines bulk deletion can now be configured for `cluster-autoscaler` via the field `.spec.kubernetes.clusterAutoscaler.maxEmptyBulkDelete` in the `Shoot` API .
```

### CODE REVIEW

File: pkg/apis/core/types_shoot.go [hunk 1]
```diff
@@ -479,6 +479,10 @@ type ClusterAutoscaler struct {
 	MaxGracefulTerminationSeconds *int32
 	// IgnoreTaints specifies a list of taint keys to ignore in node templates when considering to scale a node group.
 	IgnoreTaints []string
+	// NewPodScaleupDelay specifies how long CA should ignore newly created pods before they have to be considered for scale-up
```
Comment (shafeeqes) [inline pkg/apis/core/types_shoot.go]:
```suggestion
	// NewPodScaleupDelay specifies how long CA should ignore newly created pods before they have to be considered for scale-up.
```

File: pkg/apis/core/types_shoot.go [hunk 2]
```diff
@@ -479,6 +479,10 @@ type ClusterAutoscaler struct {
...
+	NewPodScaleupDelay *metav1.Duration
+	// MaxEmptyBulkDelete specifies the maximum number of empty nodes that can be deleted at the same time (default: 10)
```
Comment (shafeeqes) [inline pkg/apis/core/types_shoot.go]:
```suggestion
	// MaxEmptyBulkDelete specifies the maximum number of empty nodes that can be deleted at the same time (default: 10).
```

File: pkg/apis/core/v1beta1/types_shoot.go [hunk 3]
```diff
@@ -593,6 +593,12 @@ type ClusterAutoscaler struct {
 	// IgnoreTaints specifies a list of taint keys to ignore in node templates when considering to scale a node group.
 	// +optional
 	IgnoreTaints []string `json:"ignoreTaints,omitempty" protobuf:"bytes,10,opt,name=ignoreTaints"`
+	// NewPodScaleupDelay specifies how long CA should ignore newly created pods before they have to be considered for scale-up
+	// +optional
+	NewPodScaleupDelay *metav1.Duration `json:"newPodScaleupDelay,omitempty" protobuf:"bytes,11,opt,name=newPodScaleupDelay"`
+	// MaxEmptyBulkDelete specifies the maximum number of empty nodes that can be deleted at the same time (default: 10)
+	// +optional
+	MaxEmptyBulkDelete *int32 `json:"maxEmptyBulkDelete,omitempty" protobuf:"bytes,12,opt,name=maxEmptyBulkDelete"`
```
Comment (shafeeqes) [inline pkg/apis/core/v1beta1/types_shoot.go]:
```suggestion
	MaxEmptyBulkDelete *int32 `json:"maxEmptyBulkDelete,omitempty" protobuf:"varint,12,opt,name=maxEmptyBulkDelete"`
```
Please remove the protobuf tag, and run `make generate WHAT="protobuf"`. This yields a separate result compared to this PR.

**Review by shafeeqes — CHANGES_REQUESTED** [review]

File: example/90-shoot.yaml [hunk 4]
```diff
@@ -289,6 +289,8 @@ spec:
   #   ignoreTaints: 
   #     - "node.kubernetes.io/memory-pressure"
   #     - "node.kubernetes.io/disk-pressure"
+  #   newPodScaleupDelay: 10s
```
Comment (rfranzke) [inline example/90-shoot.yaml]:
```suggestion
  #   newPodScaleUpDelay: 10s
```

File: pkg/apis/core/types_shoot.go [hunk 5]
```diff
@@ -479,6 +479,10 @@ type ClusterAutoscaler struct {
 	MaxGracefulTerminationSeconds *int32
 	// IgnoreTaints specifies a list of taint keys to ignore in node templates when considering to scale a node group.
 	IgnoreTaints []string
+	// NewPodScaleupDelay specifies how long CA should ignore newly created pods before they have to be considered for scale-up
+	NewPodScaleupDelay *metav1.Duration
```
Comment (rfranzke) [inline pkg/apis/core/types_shoot.go]:
```suggestion
	NewPodScaleUpDelay *metav1.Duration
```

File: pkg/apis/core/v1beta1/types_shoot.go [hunk 6]
```diff
@@ -593,6 +593,12 @@ type ClusterAutoscaler struct {
 	// IgnoreTaints specifies a list of taint keys to ignore in node templates when considering to scale a node group.
 	// +optional
 	IgnoreTaints []string `json:"ignoreTaints,omitempty" protobuf:"bytes,10,opt,name=ignoreTaints"`
+	// NewPodScaleupDelay specifies how long CA should ignore newly created pods before they have to be considered for scale-up
+	// +optional
+	NewPodScaleupDelay *metav1.Duration `json:"newPodScaleupDelay,omitempty" protobuf:"bytes,11,opt,name=newPodScaleupDelay"`
```
Comment (rfranzke) [inline pkg/apis/core/v1beta1/types_shoot.go]:
```suggestion
	NewPodScaleUpDelay *metav1.Duration `json:"newPodScaleUpDelay,omitempty" protobuf:"bytes,11,opt,name=newPodScaleupDelay"`
```

**Review by rfranzke — COMMENTED** [review]
Extend https://github.com/gardener/gardener/blob/master/docs/usage/shoot_autoscaling.md#horizontal-node-auto-scaling

File: pkg/apis/core/v1beta1/types_shoot.go [hunk 7]
```diff
@@ -593,6 +593,12 @@ type ClusterAutoscaler struct {
 	// IgnoreTaints specifies a list of taint keys to ignore in node templates when considering to scale a node group.
 	// +optional
 	IgnoreTaints []string `json:"ignoreTaints,omitempty" protobuf:"bytes,10,opt,name=ignoreTaints"`
+	// NewPodScaleUpDelay specifies how long CA should ignore newly created pods before they have to be considered for scale-up
```
Comment (rfranzke) [inline pkg/apis/core/v1beta1/types_shoot.go]:
```suggestion
	// NewPodScaleUpDelay specifies how long CA should ignore newly created pods before they have to be considered for scale-up.
```

File: pkg/apis/core/v1beta1/types_shoot.go [hunk 8]
```diff
@@ -593,6 +593,12 @@ type ClusterAutoscaler struct {
...
+	// +optional
+	NewPodScaleUpDelay *metav1.Duration `json:"newPodScaleUpDelay,omitempty" protobuf:"bytes,11,opt,name=newPodScaleUpDelay"`
+	// MaxEmptyBulkDelete specifies the maximum number of empty nodes that can be deleted at the same time (default: 10)
```
Comment (rfranzke) [inline pkg/apis/core/v1beta1/types_shoot.go]:
```suggestion
	// MaxEmptyBulkDelete specifies the maximum number of empty nodes that can be deleted at the same time (default: 10).
```

**Review by rfranzke — COMMENTED** [review]
/lgtm

**Pull request merged**